### PR TITLE
Feature/questions

### PR DESCRIPTION
**Pull request by corysanders3**
# Pull Request

## Description
- Add a question form to each specific habit

## Related Issues
- I am unable to view the calendar due to the following error: 'cannot read properties of undefined: reading status'.

## Screenshots (if applicable)
Screenshot 2024-06-06 at 11 17 24 PM

## Checklist
- [x] The code follows the project's coding standards.
- [] Unit tests have been added or updated to cover the changes.
- [x] The code compiles without errors.
- [x] All new and existing tests pass.
- [x] The pull request has been reviewed by at least one other contributor.

### CODE REVIEW

File: src/flowerdetails/FlowerDetails.js [hunk 1]
```diff
@@ -44,6 +44,11 @@ function FlowerDetails({ details, getDetails }) {
                     <dd className="mt-1"><span className="font-semibold">End: </span> {endDate}</dd>
                 </div>
             </div>
+            <div className="font-semibold mt-0 mb-4 mx-6 text-gray-60 bg-gray-200 p-3 pl-5 pr-5 rounded-md hover:bg-lime-200 hover:cursor-pointer">
+                <button onClick={e => {
+                    loadForm(e, questionActive, setQuestionActive)
+                }}>Got A Question About This Habit?</button>
+            </div>
```
Comment (dkwon1223) [inline src/flowerdetails/FlowerDetails.js:51]: I like this implementation!

File: src/question/Question.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    const [formError, setFormError] = useState('');
+    const [result, setResult] = useState('');
+
+    function checkForm(e) {
+        e.preventDefault()
+        setFormError('')
+        setResult('')
+        
+        if(question.trim().length < 6) {
+            setFormError('Question must be at least 5 characters long.')
```
Comment (dkwon1223) [inline src/question/Question.js:16]: Nice input handling

File: src/question/Question.css [hunk 3]
```diff
@@ -0,0 +1,81 @@
+.question-section {
+    position: absolute;
+    top: 15%;
+    left: 35%;
+    z-index: 2;
+    border: 3px solid black;
+    padding: 40px 50px;
+    background-color: rgba(240, 240, 240, 0.944);
```
Comment (dkwon1223) [inline src/question/Question.css:8]: Great consistency in styling with the new habit pop up

File: src/flowerdetails/FlowerDetails.js
Comment (crotteau) [inline src/flowerdetails/FlowerDetails.js:1]: nice method for implementing the question form!

**Review by crotteau — APPROVED** [review]

**Pull request merged**